controllers/esrally/statefulset.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2019 The xridge kubestone contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package esrally
+
+import (
+	"github.com/xridge/kubestone/api/v1alpha1"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+func NewStatefulSet(cr *v1alpha1.EsRally, coordinatorHostname string) (*v1.StatefulSet, error) {
+	selectorLabels := map[string]string{
+		"perf.kubestone.xridge.io/benchmark": "esrally",
+		"perf.kubestone.xridge.io/instance":  cr.Name,
+	}
+	podLabels := map[string]string{}
+
+	//coordinatorHostname := fmt.Sprintf("%s-coordinator.%s", cr.Name, cr.Namespace)
+	//coordinatorHostname = ip
+
+	for k, v := range selectorLabels {
+		podLabels[k] = v
+	}
+
+	for k, v := range cr.Spec.PodConfig.PodLabels {
+		podLabels[k] = v
+	}
+
+	quantity, err := resource.ParseQuantity(cr.Spec.Persistence.Size)
+	if err != nil {
+		return nil, err
+	}
+
+	volumeClaims := []corev1.PersistentVolumeClaim{
```

Comment:
If we'd bring in the `volume` and `volumeSource` from fio, emptyDir based scratch space would also be feasible.